airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.integrations.destination.jdbc.typing_deduping;
+
+import io.airbyte.cdk.integrations.destination.NamingConventionTransformer;
+import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition;
+import io.airbyte.integrations.base.destination.typing_deduping.AirbyteProtocolType;
+import io.airbyte.integrations.base.destination.typing_deduping.AirbyteType;
+import io.airbyte.integrations.base.destination.typing_deduping.Array;
+import io.airbyte.integrations.base.destination.typing_deduping.ColumnId;
+import io.airbyte.integrations.base.destination.typing_deduping.SqlGenerator;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId;
+import io.airbyte.integrations.base.destination.typing_deduping.Struct;
+import io.airbyte.integrations.base.destination.typing_deduping.Union;
+import io.airbyte.integrations.base.destination.typing_deduping.UnsupportedOneOf;
+import org.jooq.DataType;
+import org.jooq.SQLDialect;
+import org.jooq.impl.SQLDataType;
+
+public abstract class JdbcSqlGenerator implements SqlGenerator<TableDefinition> {
+
+  private final NamingConventionTransformer namingTransformer;
+
+  public JdbcSqlGenerator(final NamingConventionTransformer namingTransformer) {
+    this.namingTransformer = namingTransformer;
+  }
+
+  @Override
+  public StreamId buildStreamId(final String namespace, final String name, final String rawNamespaceOverride) {
+    return new StreamId(
+        namingTransformer.getNamespace(namespace),
+        namingTransformer.convertStreamName(name),
+        namingTransformer.getNamespace(rawNamespaceOverride),
+        namingTransformer.convertStreamName(StreamId.concatenateRawTableName(namespace, name)),
+        namespace,
+        name);
+  }
+
+  public ColumnId buildColumnId(final String name, final String suffix) {
+    final String nameWithSuffix = name + suffix;
+    return new ColumnId(
+        namingTransformer.getIdentifier(nameWithSuffix),
+        name,
+        namingTransformer.getIdentifier(nameWithSuffix));
+  }
+
+  protected DataType<?> toDialectType(final AirbyteType type) {
+    if (type instanceof final AirbyteProtocolType airbyteProtocolType) {
+      return toDialectType(airbyteProtocolType);
+    }
+    return switch (type.getTypeName()) {
```

Comment:
I can't wait for us to upgrade java and get access to pattern matching switch statements >.>